torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -61,12 +62,42 @@ TORCH_CUDA_API Val* promoteNew(Val* v1, Val* v2) {
   return newValLike(v1, out_dtype);
 }
 
+Val* newConstScalar(DataType dtype, long int val) {
+  switch (dtype) {
+    case (DataType::Int):
+      return new Int(val);
+    default:
+      break;
+  }
+  TORCH_CHECK(
+      false,
+      "Could not generate a new Scalar with data type ",
+      dtype,
+      "and constant value: ",
+      val);
+}
+
+Val* newConstScalar(DataType dtype, double val) {
```

Comment:
Consider changing function signature, since it returns `Float`
```suggestion
Val* newConstScalar(DataType dtype, float val) {
```